include/nlohmann/detail/conversions/to_json.hpp

```diff
@@ -9,30 +9,34 @@
 #include <valarray> // valarray
 #include <vector> // vector
 
-#include <nlohmann/detail/macro_scope.hpp>
 #include <nlohmann/detail/iterators/iteration_proxy.hpp>
+#include <nlohmann/detail/macro_scope.hpp>
 #include <nlohmann/detail/meta/cpp_future.hpp>
 #include <nlohmann/detail/meta/type_traits.hpp>
 #include <nlohmann/detail/value_t.hpp>
 
 #if JSON_HAS_EXPERIMENTAL_FILESYSTEM
 #include <experimental/filesystem>
-namespace nlohmann::detail
+NLOHMANN_JSON_NAMESPACE_BEGIN
+namespace detail
 {
 namespace std_fs = std::experimental::filesystem;
-} // namespace nlohmann::detail
+}  // namespace detail
+NLOHMANN_JSON_NAMESPACE_END
 #elif JSON_HAS_FILESYSTEM
 #include <filesystem>
-namespace nlohmann::detail
+NLOHMANN_JSON_NAMESPACE_BEGIN
+namespace detail
 {
 namespace std_fs = std::filesystem;
-} // namespace nlohmann::detail
+}  // namespace detail
+NLOHMANN_JSON_NAMESPACE_END
```

Comment:
Just thinking out loud, it would be nice if we could avoid having to duplicate this block of #if/#elif in multiple places.  Not saying it needs to be in this PR.